packages/core/admin/admin/src/hooks/useElementOnScreen.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+
+/**
+ * Hook that returns a ref to an element and a boolean indicating if the element is in the viewport
+ * or in the element specified in `options.root`.
+ */
+const useElementOnScreen = <TElement extends HTMLElement = HTMLElement>(
+  callback: IntersectionObserverCallback,
+  options?: IntersectionObserverInit
+): React.RefObject<TElement> => {
+  const containerRef = React.useRef<TElement>(null);
+  const id = React.useId();
+
+  React.useEffect(() => {
+    const containerEl = containerRef.current;
+    const observer = new IntersectionObserver(callback, options);
```

Comment:
I'm considering abstracting the callback further, so the hook would take onIsVisibleChange. Less flexible but easier to use and understand 🤔

Reply:
I'd be interested to hear more of your reasoning and discuss it but I think the way you have it makes sense.

Reply:
When writing the callback I realized that entry.isIntersecting is weird, it's not clear if it means it's visible or not. Since the hook is useElementOnScreen and not useIntersectionObserver I find it weird to deal with something low-level like this.

So I think I'll change it, if we need more control somewhere else in the future we could always call IntersectionObserver directly